### ISSUE TITLE
RunCommand v2 API

### ISSUE THREAD

**Issue by koralski** [Azure/azure-cli#19319]
**Resource Provider**

Microsoft.Compute

**Description of Feature or Work Requested**

Implement CLI cmdlets for Runcommand REST API

**Minimum API Version Required**

2021-07-01

**Swagger Link**

https://github.com/Azure/azure-rest-api-specs/blob/main/specification/compute/resource-manager/Microsoft.Compute/stable/2021-07-01/runCommands.json

**Target Date**
Public preview release: October 2021

**Comment by yonzhan** [comment]
Compute

**Comment by zhoxing-ms** [comment]
@koralski  Since we have several important requirements to do in the near future, so we plan to make the development of this requirement to the sprint in October (it will be released in **11-02**). Do you think the release time is acceptable?

**Comment by koralski** [comment]
> @koralski Since we have several important requirements to do in the near future, so we plan to make the development of this requirement to the sprint in October (it will be released in **11-02**). Do you think the release time is acceptable?

This feature is currently in private preview and public preview is scheduled for end of October 2021. Only release blocking items are Powershell and CLI support. Se we prefer to have CLI ready as soon as possible aligned with your release. You can sync up with our PM: amjad.shaik@microsoft.com on the schedule

**Comment by zhoxing-ms** [comment]
@koralski  @amjads1 I'd like to confirm with you that this requirement requires CLI to support the following new commands, right?

>az vm run-command delete 
az vm run-command list
az vm run-command set
az vm run-command show 
az vm run-command wait 
az vmss run-command delete
az vmss run-command list
az vmss run-command set
az vmss run-command show

Is there anything else we need to support?

**Comment by koralski** [comment]
> @koralski @amjads1 I'd like to confirm with you that this requirement requires CLI to support the following new commands, right?
> 
> > az vm run-command delete
> > az vm run-command list
> > az vm run-command set
> > az vm run-command show
> > az vm run-command wait
> > az vmss run-command delete
> > az vmss run-command list
> > az vmss run-command set
> > az vmss run-command show
> 
> Is there anything else we need to support?

Yes. These are the new APIs. Refer to the spec link above for details.
